flopy/mf6/data/mfdataplist.py

```diff
@@ -0,0 +1,2527 @@
+            )
+
+        data_storage = self._get_storage()
+        if append:
+            # append data to existing dataframe
+            current_data = self._get_dataframe()
+            if current_data is not None:
+                data = pandas.concat([current_data, data])
+        if data_storage.data_storage_type == DataStorageType.external_file:
+            # store external data until next write
+            data_storage.internal_data = data
+        else:
+            # store data internally
+            data_storage.set_internal(data)
+            data_storage.modified = True
+
+    def has_modified_ext_data(self):
+        """check to see if external data has been modified since last read"""
+        data_storage = self._get_storage()
+        return (
+            data_storage.data_storage_type == DataStorageType.external_file
+            and data_storage.internal_data is not None
+        )
+
+    def binary_ext_data(self):
+        """check for binary data"""
+        data_storage = self._get_storage()
+        return data_storage.binary
+
+    def to_array(self, kper=0, mask=False):
+        """Convert stress period boundary condition (MFDataList) data for a
+        specified stress period to a 3-D numpy array.
+
+        Parameters
+        ----------
+        kper : int
+            MODFLOW zero-based stress period number to return (default is
+            zero)
+        mask : bool
+            return array with np.NaN instead of zero
+
+        Returns
+        ----------
+        out : dict of numpy.ndarrays
+            Dictionary of 3-D numpy arrays containing the stress period data
+            for a selected stress period. The dictionary keys are the
+            MFDataList dtype names for the stress period data."""
+        sarr = self.get_data(key=kper)
+        model_grid = self._data_dimensions.get_model_grid()
+        return list_to_array(sarr, model_grid, kper, mask)
+
+    def set_data(self, data, autofill=False, check_data=True):
+        """Sets the contents of the data to "data".  Data can have the
+        following formats:
+            1) recarray - recarray containing the datalist
+            2) [(line_one), (line_two), ...] - list where each line of the
+               datalist is a tuple within the list
+        If the data is transient, a dictionary can be used to specify each
+        stress period where the dictionary key is <stress period> - 1 and
+        the dictionary value is the datalist data defined above:
+        {0:ndarray, 1:[(line_one), (line_two), ...], 2:{'filename':filename})
+
+        Parameters
+        ----------
+            data : ndarray/list/dict
+                Data to set
+            autofill : bool
+                Automatically correct data
+            check_data : bool
+                Whether to verify the data
+
+        """
+        self._set_data(data, check_data=check_data)
+
+    def set_record(self, data_record, autofill=False, check_data=True):
```

Comment:
Variable name changed to be more consistent with method name.  Changed variable name instead of method name since method name change would breaking existing interface.

Having the hidden method is not necessary.  I was originally doing this to make sure the correct method (parent vs child class) got called.  But it is better to just explicitly define this, which I am now doing.